sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/ForeachSinkSuite.scala

```diff
@@ -204,6 +204,55 @@ class ForeachSinkSuite extends StreamTest with SharedSQLContext with BeforeAndAf
       query.stop()
     }
   }
+
+  test("foreach with watermark: append") {
```

Comment:
what does this test that is not covered in the previous test "watermark + complete"?